build.gradle.kts

```diff
@@ -0,0 +1,119 @@
+allprojects {
+    group = "com.amazonaws"
+
+    repositories {
+        mavenCentral()
+        mavenLocal()
+    }
+
+    plugins.withId("java-library") {
+        configure<JavaPluginExtension> {
+            sourceCompatibility = JavaVersion.VERSION_1_8
+            targetCompatibility = JavaVersion.VERSION_1_8
+
+            withJavadocJar()
+            withSourcesJar()
+        }
+
+        val propertiesDir = "build/generated/sources/properties"
+
+        configure<JavaPluginConvention> {
+            sourceSets {
+                named("main") {
+                    output.dir(mapOf("builtBy" to "generateProperties"), propertiesDir)
+                }
+            }
+        }
+
+        dependencies {
+            configurations.configureEach {
+                if (isCanBeResolved && !isCanBeConsumed) {
+                    add(name, platform(project(":dependencyManagement")))
+                }
+            }
+        }
+
+        tasks {
+            named<Javadoc>("javadoc") {
+                val options = options as StandardJavadocDocletOptions
+
+                options.quiet()
+                options.addBooleanOption("-allow-script-in-comments", true)
+                options.footer = "<script src=\"/SdkStatic/sdk-priv.js\" async=\"true\"></script>"
+                options.bottom = "Copyright &#169; 2018 Amazon Web Services, Inc. All Rights Reserved."
+
+                // TODO(anuraaga): Enable doclint except for required @param/@returns
+                options.addBooleanOption("Xdoclint:none", true)
+                // options.addBooleanOption("Xdoclint:accessibility", true)
+                // options.addBooleanOption("Xdoclint:html", true)
+                // options.addBooleanOption("Xdoclint:reference", true)
+                // options.addBooleanOption("Xdoclint:syntax", true)
+            }
+
+            val generateProperties by registering {
+                doLast {
+                    val folder = file("${propertiesDir}/com/amazonaws/xray")
+                    folder.mkdirs()
+                    val propertiesFile = folder.resolve("sdk.properties")
+                    propertiesFile.writeText("awsxrayrecordersdk.version=${project.version}")
+                }
+            }
+        }
+    }
+
+    plugins.withId("maven-publish") {
```

Comment:
Where is GPG signing done? Do we need to configure the target repo for publishing like we do [here](https://github.com/aws/aws-xray-sdk-java/blob/master/pom.xml#L74-L80)?

Reply:
Good eye - I was thinking of following up with verifying the config and commands for releasing but at least adding the publishing repo now is straightforward so added it.

Reply:
Fair enough, since we skip signing for non-release builds anyway. We'll come back to this.

Reply:
For reference, I remember there was some issues about signing in Gradle with the X-Ray OT SDK: https://github.com/awslabs/aws-xray-sdk-with-opentelemetry/blob/master/build.gradle#L80-L89

But as per the linked GitHub issue, it looks to be resolved with Gradle 6.2 so we should be good.

Reply:
Thanks for this pointer :)